subprojects/tooling-api/src/main/java/org/gradle/tooling/model/build/BuildEnvironment.java

```diff
@@ -65,4 +67,12 @@
      * @throws UnsupportedMethodException For Gradle versions older than 1.0-milestone-8, where this method is not supported.
      */
     JavaEnvironment getJava() throws UnsupportedMethodException;
+
+    /**
+     * Returns an unmodifiable map of the current system environment variables.
+     *
+     * @return the environment as a map of variable names to values
+     * @since 3.5
+     */
+    Map<String, String> getEnvironmentVariables();
```

Comment:
You're right.
Initially, the idea was to add the ability to query env vars of already running daemon.
But, it's not possible (and maybe not viable) using the current design:

1. There is an optimization for the `BuildEnvironment `model => building w/o launching the daemon.  Btw, it looks like `BuildEnvironmentBuilder `is not used at all.
2. Even w/o the previous optimization, querying of the `BuildEnvironment `model will be also `LongRunningOperation `which should be run by the daemon with the same env vars as requested.

So, I've removed the method and changed integration tests accordingly.